Classes/Controller/Form.php

```diff
@@ -1109,6 +1109,9 @@ protected function init()
 
         $this->gp = $this->utilityFuncs->getMergedGP();
 
+        if (!$this->settings['uniqueFormID']) {
```

Comment:
Maybe move the `preg_replace` to line 1120 and sanitize `$randomID` unconditionally.